webrtc/sctp/sctp_utils.nim

```diff
@@ -0,0 +1,82 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import nativesockets
+import binary_serialization, chronos
+
+var errno* {.importc, header: "<errno.h>".}: cint ## error variable
+
+when defined(windows):
+  import winlean
+  const
+    SctpAF_INET* = winlean.AF_INET
+    SctpEINPROGRESS* = winlean.WSAEINPROGRESS.cint
+else:
+  const
+    SctpAF_INET* = nativesockets.AF_INET
+    SctpEINPROGRESS* = chronos.EINPROGRESS.cint
+
+type
+  # These three objects are used for debugging/trace only
+  SctpChunk* = object
+    chunkType*: uint8
+    flag*: uint8
+    length* {.bin_value: it.data.len() + 4.}: uint16
+    data* {.bin_len: it.length - 4.}: seq[byte]
+
+  SctpPacketHeader* = object
+    srcPort*: uint16
+    dstPort*: uint16
+    verifTag*: uint32
+    checksum*: uint32
+
+  SctpPacketStructure* = object
+    header*: SctpPacketHeader
+    chunks*: seq[SctpChunk]
+
+proc dataToString(data: seq[byte]): string =
+  # Only used for debugging/trace
```

Comment:
By "debugging/trace" do you mean logging? If so, it's probably better to move them to another file or rename this one to something like `logutils`. Is it a pattern to use underscore for filenames in nim? I think it's the first time I see this in our codebase.

Reply:
Done.
I don't know if it's a pattern to use underscore for filenames, but there's at least three files with this pattern in libp2p so I guess it's not unusual.